src/Confluent.Kafka.Avro/AvroSerializer.cs

```diff
@@ -88,48 +109,37 @@ public AvroSerializer(ISchemaRegistryClient schemaRegistryClient)
         /// <param name="data">
         ///     The object to serialize.
         /// </param>
+        /// <param name="isKey">
+        ///     whether or not the data represents a message key.
+        /// </param>
+        /// <param name="schemaRegistryClient">
+        ///     An implementation of ISchemaRegistryClient used for
+        ///     communication with Confluent Schema Registry.
+        /// </param>
         /// <returns>
         ///     <paramref name="data" /> serialized as a byte array.
         /// </returns>
-        public byte[] Serialize(string topic, T data)
+        public async Task<byte[]> Serialize(ISchemaRegistryClient schemaRegistryClient, string topic, T data, bool isKey)
```

Comment:
My intent was that `AvroProducer` be very specific to schema registry. I think having an `ISchemaRetriever` abstraction is going too far, but your comment makes me think `AvroProducer`/`Consumer` may not be the best name as schema registry will almost certainly support other serialization formats in the future and it seems it will make sense to incorporate that functionality into the existing classes here.

Reply:
Rather than having `AvroProducer`/`AvroConsumer`, I'm now thinking it would definitely be better to have clients tied to the serde types - especially since we can foresee Schema Registry working with more than avro. Currently have this in a local branch and currently call them called `RegisteringProducer` and `RegisteringConsumer`, though those names aren't quite appropriate.

In this case, it's appropriate for the `SchemaRegistryClient` instance to come via the serializer (not client), as @rnpridgeon questions / suggests.